scalameta/semantic/src/main/scala/scala/meta/semantic/v1/Database.scala

```diff
@@ -35,77 +31,10 @@ case class Database(
     })
     buf.toString.trim + EOL
   }
-
-  def toBinary: Array[Byte] = {
-    // TODO: This is obviously a very naive implementation, optimized for easy eyeballing.
-    // In the future, I'd like to have something that takes less space and is more performant.
-    toString.getBytes(Charset.forName("UTF-8"));
-  }
-
-  def toFile(file: File): Unit = {
-    val fos = new FileOutputStream(file)
-    try fos.write(toBinary)
-    finally { fos.close() }
-  }
 }
 
 object Database extends DatabaseOps {
   def apply(): Database = {
     Database(Map[Location, Symbol]())
   }
-
-  def apply(bytes: Array[Byte]): Database = {
-    val symbols = mutable.Map[Location, Symbol]()
-    val lines = new String(bytes, Charset.forName("UTF-8")).split(EOL)
-    var i = 0
-
-    def readSection(): Map[Location, Symbol] = {
-      val section = mutable.Map[Location, Symbol]()
-      val header = lines(i)
-      i += 1
-
-      while (i < lines.length && lines(i) != "") {
-        // [7..11): Test => _empty_.Test.
-        val Mapping = """^\[(\d+)\.\.(\d+)\): .*? => (.*)$""".r
-        val line = lines(i)
-        i += 1
-        try {
-          line match {
-            case Mapping(s_start, s_end, s_id) =>
-              val location = Location(Address(header), s_start.toInt, s_end.toInt)
-              val symbol = Symbol(s_id)
-              section(location) = symbol
-            case _ =>
-              sys.error("unexpected line format")
-          }
-        } catch {
-          case ex: Exception =>
-            sys.error(s"$i:${ex.getMessage}$EOL$line")
-        }
-      }
-      i += 1
-
-      section.toMap
-    }
-
-    while (i < lines.length) {
-      val section = readSection()
-      symbols.retain((k, _) => k.addr != section.head._1.addr)
-      symbols ++= section
-    }
-
-    Database(symbols.toMap)
-  }
-
-  def apply(file: File): Database = {
-    val fis = new FileInputStream(file)
-    try {
-      val length = fis.available
-      val bytes = new Array[Byte](length)
-      fis.read(bytes)
-      Database(bytes)
-    } finally {
-      fis.close()
-    }
-  }
```

Comment:
Oh I got it. I mean I suggest to remove the old text-based logic and replace it with the new pb-based logic, so that scalameta/semantic can serialize/deserialize via protobuf.